src/parseexpr.jl

```diff
@@ -261,14 +261,35 @@ function destructive_add!(ex::AbstractArray{T}, c::Number, x::AbstractArray) whe
     add_to_expression!.(ex, c*x)
 end
 
+function destructive_add!(ex::AbstractArray{<:GenericAffExpr},
+                          c::AbstractArray{<:GenericQuadExpr},
+                          x::Number)
+    result = c*x
+    add_to_expression!.(result, ex)
+    return result
+end
+function destructive_add!(ex::AbstractArray{<:GenericAffExpr}, c::Number,
+                          x::AbstractArray{<:GenericQuadExpr})
+    result = c*x
+    add_to_expression!.(result, ex)
+    return result
+end
+
+
 destructive_add!(ex, c, x) = ex + c*x
```

Comment:
I don't understand the 5 different fallbacks right here. What is going on and why are tuples needed?

Reply:
These were added in https://github.com/JuliaOpt/JuMP.jl/pull/1416 by @mohamed82008 
I suspect it is for ones using the `(...,)` syntax is to make a scalar object that does not implement `Base.broadcastable` broadcastable but a simpler fix might be to define `broadcastable` for `JuMPScalar` elements

Reply:
True.

Reply:
I added an issue: https://github.com/JuliaOpt/JuMP.jl/issues/1430 as it might need some bikeshedding which will be clearer if done in a separate PR